src/code/matcher/matcher.cpp

```diff
@@ -69,13 +69,48 @@ const IServerConf *matcher::requestToServer(const IConf *conf, const IListen *so
 const ILocation *matcher::requestToLocation(const IServerConf *s, const BasicHTTPRequest &req) throw()
 {
 	std::list<const ILocation *> list = s->getLocations();
+	std::list<const ILocation *>::const_iterator tmp = list.end();
 	std::list<const ILocation *>::const_iterator it = list.begin();
-
 	while (it != list.end())
 	{
-		if ((*it)->getPath().get() == req.getPath())
-			return *it;
-		++it;
+		if ((req.getPath().find((*it)->getPath().get()) == 0) || (req.getPath() == "" && (*it)->getPath().get() == "/"))
+		{
+			if ((tmp == list.end()) || ((*it)->getPath().get().length() > (*tmp)->getPath().get().length()))
+				tmp = it;
+		}
+		it++;
 	}
-	return NULL;
+	if (tmp == list.end())
+		return NULL;
+	return (*tmp);
+}
+
+Path matcher::rootToRequest(const Server *server, const BasicHTTPRequest &req)
```

Comment:
the HTTPCall object should have both ServerConf and Location. we should use them here:
```
Path matcher::rootToRequest(const HTTPCall &call)
{
	const ILocation *loc = requestToLocation(server->getConf(), req);
	if (call.location != NULL)
	{
		return (Path(call.location->getRoot()->get()));
	}
	return (Path(call.server->getRoot()->get()));
}
```